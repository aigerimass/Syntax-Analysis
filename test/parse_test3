F(){
    return 4;
}
Func_with_While(condition){
    while(condition > 0){
        a = a + 5;
    };
    return a + 4*8^2;
}
Main(){
Func_with_While(1);
skip;
}
