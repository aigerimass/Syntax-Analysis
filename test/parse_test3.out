----------SYNTAX-ANALYSIS----------
Function Definition: [
name = "F",
args: 
body:
>Return(4);

]
--------
Function Definition: [
name = "Func_with_While",
args: condition
body:
>While(
cond: 
Gt(Var "condition", 0)
body:
>Bind(Var "a", Plus(Var "a", 5));
);

>Return(Plus(Var "a", Mult(4, Pow(8, 2))));

]
--------
Function Definition: [
name = "Main",
args: 
body:
>Call(name: "Func_with_While", args: 1);

Skip();

]
--------
----------INTERVAL-ANALYSIS----------
ERROR: VARIABLE "a" doesn't exist
